Doc/reference/expressions.rst

```diff
@@ -595,12 +595,19 @@ is already executing raises a :exc:`ValueError` exception.
 .. method:: generator.close()
 
    Raises a :exc:`GeneratorExit` at the point where the generator function was
-   paused.  If the generator function then exits gracefully, is already closed,
-   or raises :exc:`GeneratorExit` (by not catching the exception), close
-   returns to its caller.  If the generator yields a value, a
-   :exc:`RuntimeError` is raised.  If the generator raises any other exception,
-   it is propagated to the caller.  :meth:`close` does nothing if the generator
-   has already exited due to an exception or normal exit.
+   paused.  If the generator function then returns a value (by catching
+   :exc:`GeneratorExit`), it is returned by :meth:`close`.  If the generator
```

Comment:
I found it confusing that it "returns a value (by catching an exception)".   How about:

If the generator function catches the exception and returns a value, this value is returned from :meth:`close`.